src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -79,94 +70,52 @@ public HostOutputMessage HostOutputMessage
         {
-            bool ownsInstance = false;
-
-            if (!(instance is IFunctionInstanceEx functionInstance))
-            {
-                functionInstance = new FunctionInstanceWrapper(instance, _serviceScopeFactory);
-                ownsInstance = true;
-            }
-
-            try
-            {
-                return await TryExecuteAsyncCore(functionInstance, cancellationToken);
-            }
-            finally
-            {
-                if (ownsInstance)
-                {
-                    (functionInstance as IDisposable)?.Dispose();
-                }
-            }
-        }
-
-        private async Task<IDelayedException> TryExecuteAsyncCore(IFunctionInstanceEx functionInstance, CancellationToken cancellationToken)
-        {
-            ILogger logger = _loggerFactory?.CreateLogger(LogCategories.CreateFunctionCategory(functionInstance.FunctionDescriptor.LogName));
-
-            FunctionStartedMessage functionStartedMessage = CreateStartedMessageWithoutArguments(functionInstance);
-            var parameterHelper = new ParameterHelper(functionInstance);
-            FunctionCompletedMessage functionCompletedMessage = null;
+            var functionInstanceEx = (IFunctionInstanceEx)functionInstance;
+            var logger = _loggerFactory?.CreateLogger(LogCategories.CreateFunctionCategory(functionInstance.FunctionDescriptor.LogName));
+            var functionStartedMessage = CreateStartedMessage(functionInstanceEx);
+            var parameterHelper = new ParameterHelper(functionInstanceEx);
             ExceptionDispatchInfo exceptionInfo = null;
             string functionStartedMessageId = null;
             FunctionInstanceLogEntry instanceLogEntry = null;
 
-            using (_resultsLogger?.BeginFunctionScope(functionInstance, HostOutputMessage.HostInstanceId))
+            using (_resultsLogger?.BeginFunctionScope(functionInstanceEx, HostOutputMessage.HostInstanceId))
             using (parameterHelper)
             {
                 try
                 {
                     instanceLogEntry = await NotifyPreBindAsync(functionStartedMessage);
                     parameterHelper.Initialize();
-                    functionStartedMessageId = await ExecuteWithLoggingAsync(functionInstance, functionStartedMessage, instanceLogEntry, parameterHelper, logger, cancellationToken);
-                    functionCompletedMessage = CreateCompletedMessage(functionStartedMessage);
+                    functionStartedMessageId = await ExecuteWithLoggingAsync(functionInstanceEx, functionStartedMessage, instanceLogEntry, parameterHelper, logger, cancellationToken);
                 }
                 catch (Exception exception)
                 {
-                    if (functionCompletedMessage == null)
-                    {
-                        functionCompletedMessage = CreateCompletedMessage(functionStartedMessage);
```

Comment:
Removed this object creation, which was just a copy of all the data that was already in the started message. A FunctionCompletedMessage "isa" FunctionStartedMessage, so above when creating the started message I just create an instance of the latter.